File: 🟢 Easy/884 - Uncommon Words from Two Sentences/_884UncommonWordsfromTwoSentences.js
/**
 * @param {string} s1
 * @param {string} s2
 * @return {string[]}
 */
var uncommonFromSentences = function (s1, s2) {
    let s = s1 + " " + s2;
    let arr = s.split(" ");
    let map = new Map();

    for (let str of arr) {
        if (map.has(str)) {
            map.set(str, map.get(str) + 1);
        } else {
            map.set(str, 1);
        }
    }

    let res = [];
    map.forEach((value, key) => {
        if (value == 1) {
            res.push(key);
        }
    });

    return res;
};

File: 🟢 Easy/884 - Uncommon Words from Two Sentences/_884UncommonWordsfromTwoSentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        s = s1 + " " + s2
        arr = list(s.split(" "))
        map = {}

        for str in arr:
            if str in map:
                map[str] += 1
            else:
                map[str] = 1

        res = []

        for key in map:
            if map[key] == 1:
                res.append(key)

        return res
